exams/tasks.py

```diff
@@ -2,56 +2,128 @@
 from datetime import datetime
+import logging
 import tempfile
 
+from django.conf import settings
 from django.db import transaction
 import pytz
 
-from exams.models import ExamProfile
-from exams.pearson import (
-    write_cdd_file,
-    upload_tsv,
+from exams.exceptions import RetryableSFTPException
+from exams.models import (
+    ExamAuthorization,
+    ExamProfile,
 )
+from exams import pearson
 from micromasters.celery import async
 
+PEARSON_CDD_FILE_PREFIX = "cdd-%Y%m%d%H_"
+PEARSON_EAD_FILE_PREFIX = "ead-%Y%m%d%H_"
 
-@async.task
-def export_exam_profiles():
+PEARSON_FILE_EXTENSION = ".dat"
+
+PEARSON_FILE_ENCODING = "utf-8"
+
+log = logging.getLogger(__name__)
+
+
+def _backoff(retries):
+    """
+    Exponential backoff for retried tasks
+    """
+    return settings.EXAMS_SFTP_BACKOFF_BASE ** retries
+
+
+@async.task(bind=True, max_retries=3)
+def export_exam_profiles(self):
+    """
+    Sync any outstanding profiles
+    """
+    exam_profiles = ExamProfile.objects.filter(
+        status=ExamProfile.PROFILE_PENDING).select_related('profile')
+    file_prefix = datetime.now(pytz.utc).strftime(PEARSON_CDD_FILE_PREFIX)
+
+    try:
+        # write the file out locally
+        # this will be written out to a file like: /tmp/cdd-20160405_kjfiamdf.dat
+        with tempfile.NamedTemporaryFile(
+            prefix=file_prefix,
+            encoding=PEARSON_FILE_ENCODING,
+            suffix=PEARSON_FILE_EXTENSION,
+            mode='w',
+        ) as tsv:
+            cdd_writer = pearson.CDDWriter()
+            valid_profiles, invalid_profiles = cdd_writer.write(tsv, exam_profiles)
+
+            # flush data to disk before upload
+            tsv.flush()
+
+            # upload to SFTP server
+            pearson.upload_tsv(tsv.name)
+    except RetryableSFTPException as exc:
+        log.exception('Retryable error during upload of CDD file to Pearson SFTP')
+        # retry up to 3 times w/ exponential backoff if this was a connection error
+        self.retry(exc=exc, countdown=_backoff(self.request.retries))
+    except:  # pylint: disable=bare-except
+        log.exception('Unexpected exception uploading CDD file')
+        return
+
+    try:
+        with transaction.atomic():
+            # update records to reflect the successful upload
+            if valid_profiles:
+                exam_profile_ids = [exam_profile.id for exam_profile in valid_profiles]
+                ExamProfile.objects.filter(
+                    id__in=exam_profile_ids).update(status=ExamProfile.PROFILE_IN_PROGRESS)
+
+            # update records to reflect invalid profile
+            if invalid_profiles:
+                exam_profile_ids = [exam_profile.id for exam_profile in invalid_profiles]
+                ExamProfile.objects.filter(
+                    id__in=exam_profile_ids).update(status=ExamProfile.PROFILE_INVALID)
+    except:  # pylint: disable=bare-except
+        log.exception('Unexpected exception updating ExamProfile.status')
```

Comment:
like in another comment, please wrap with `try/except` only the lines you want to handle.